packages/cosmos/src/client.rs

```diff
@@ -67,18 +69,23 @@ impl CosmosBuilders {
 }
 
 #[async_trait]
-impl deadpool::managed::Manager for CosmosBuilders {
-    type Type = CosmosInner;
+impl ManageConnection for CosmosBuilders {
+    type Connection = CosmosInner;
 
     type Error = anyhow::Error;
 
-    async fn create(&self) -> Result<Self::Type> {
+    async fn connect(&self) -> Result<Self::Connection> {
         self.get_next_builder().build_inner().await
     }
 
-    async fn recycle(&self, _: &mut CosmosInner) -> RecycleResult<anyhow::Error> {
+    async fn is_valid(&self, _: &mut CosmosInner) -> Result<()> {
+        // We do the validity check within the perform_query calls and update is_broken
```

Comment:
Given that this function is used internally by the pool library (Ref: https://github.com/djc/bb8/blob/6908d2f894dd2ed837538c0bb0fd06c2a37d5928/bb8/src/inner.rs#L125), do you think something like this is better:

```
if inner.is_broken { Ok()} else anyhow!("Some failure:)
```

Reply:
Yeah, that makes sense, I'll update it. I would have thought it would check is_broken right there too based on the docs.